### PR TITLE
MAINT-25835: Configure newly created Wiki settings and pages permissions

### PR DESCRIPTION
**Pull request by azayati**
This is a quick win in order to configure correctly the newly created wiki settings and pages permissions according to [this specification](https://community.exoplatform.com/portal/g/:spaces:platform_4/exo_platform/wiki/Wiki_Permissions)

### CODE REVIEW

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: wiki-service/src/main/java/org/exoplatform/wiki/jpa/dao/WikiDAO.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
   public List<Long> findAllIds(int offset, int limit) {
     return getEntityManager().createNamedQuery("wiki.getAllIds").setFirstResult(offset).setMaxResults(limit).getResultList();
   }
-
+
```
[the remark this answers is not in the document]
Comment (azayati) [reply wiki-service/src/main/java/org/exoplatform/wiki/jpa/dao/WikiDAO.java]: done

File: wiki-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java [hunk 2]
```diff
@@ -1441,6 +1447,50 @@ public PageEntity fetchPageEntity(Page page) {
...
+            PermissionEntry portalPermissionEntry = new PermissionEntry(portalConfig.getEditPermission(),
+                    "", IDType.MEMBERSHIP, permissions);
+            permissionEntries.add(portalPermissionEntry);
+          }
+        }
+      } catch (Exception e) {
+        throw new WikiException("Cannot get user portal config for wiki " + wikiType + ":" + wikiOwner
+                + " - Cause : " + e.getMessage(), e);
+      }
+      PermissionEntry anyPermissionEntry = new PermissionEntry(IdentityConstants.ANY, "", IDType.USER, new Permission[] {(
```
[the remark this answers is not in the document]
Comment (azayati) [reply wiki-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:1480]: Any means that any member of platform/users have access to the wiki of the private site, it is logical no ?

File: wiki-service/src/main/java/org/exoplatform/wiki/service/WikiService.java [hunk 3]
```diff
@@ -589,6 +589,13 @@ public default Attachment getAttachmentOfPageByName(String attachmentName, Page
    */
   public List<SpaceBean> searchSpaces(String keyword) throws WikiException;
   
+  /**
+   * Gets all wikis
+   * @return All wikis
+   * @throws WikiException
+   */
+  public List<Wiki> getAllWikis() throws WikiException;
```
[the remark this answers is not in the document]
Comment (azayati) [reply wiki-service/src/main/java/org/exoplatform/wiki/service/WikiService.java:597]: It is added in order to be used by the UP, I will remove it then

[the remark this answers is not in the document]
Comment (azayati) [reply wiki-service/src/main/java/org/exoplatform/wiki/service/WikiService.java]: done

File: wiki-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahamdi) [reply wiki-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:1471]: I thought Any means anyone even guest users, isn't it the case ?

[the remark this answers is not in the document]
Comment (ahamdi) [reply wiki-service/src/main/java/org/exoplatform/wiki/jpa/JPADataStorage.java:1471]: OK I think we cna proceed with the merge

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**